rancher-nfs: rework default/custom opts, always pool by default

LLParse opened the pull request:
https://github.com/rancher/rancher/issues/7906

LLParse commented on package/nfs/rancher-nfs:
```diff
@@ -73,116 +73,100 @@ create() {
-        if [ $? -ne 0 ]; then
-            print_error "Failed umount ${MNT_DEST}: ${error}"
-        fi
+delete() {
```
I haven't tested this function because Rancher doesn't seem to execute the `delete` anymore (this is acceptable to me because it is far safer to make user manually delete data - just like PVs in a K8s StatefulSet). 

It would be nice if we could `docker volume rm <name>` stack/container scoped volumes, but apparently the volume reference is deleted so docker/rancher have no knowledge of the volume after stack deletion.

Fair warning!

cjellick commented:
@LLParse I don't how much bandwidth you have, but I am hesitant to merge such a PR without any tests. We've worked ourselves into a corner by not having tests for our verified storage drivers. I'd like to figure out how to work ourselves out of that

LLParse commented:
I am booked up for the next few weeks - I would agree the lack of test cases for the storage drivers (and especially the framework) is troublesome. Maybe @StrongMonkey has some spare cycles?

cjellick commented:
@LLParse can you provide a better explanation of what you changed in this PR? The bug you referenced seem pretty targeted, but clearly more than just addressing that bug was changed in this PR.

Also, before we merge, can you build this into a custom image and work with @sangeethah to run the validation tests against it and to add a new test covering the bug you addressed; I'll leave it to you and sangeetha to figure out who should write that test, but ideally I'd like to see a passing test for it before we merge this feature.

cjellick commented on package/nfs/rancher-nfs:
```diff
@@ -1,70 +1,70 @@
 
-unmount_default_share() {
-    local mntDest=$1
+mount_default() {
```
Where does this get called and what is it used for? I couldnt find any references to it, but im not sure if it is being called in some magical frameworky way

LLParse replied:
It's dead code now

LLParse commented:
@cjellick 
When creating/mounting NFS volumes with the default (template questions) NFS configuration, we create a subdirectory for each volume. The exported directory may be used for multiple volumes.

When creating/mount NFS volumes with custom (driver_opts) NFS configuration, we do not create a subdirectory for each volume; instead, we mount the root export directory.

The bug/feature was to change the behavior to do the subdirectory creation, regardless of where the NFS configuration originated. Some users might also want to mount the root export directory (because reasons) so we introduced `pool=false` as a driver_opt to disable the subdirectory creation.

The bug was likely introduced due to poor code reuse. I refactored the logic to use the same functions for the pooling/non-pooling behaviors so we do not have to maintain 2 copies of the same logic.

As for writing tests, I am all for it. These tests should be written against master (not this PR) to establish a baseline, and then re-tested against these modifications.

Edit: Ironically, the code that was merged causing conflicts https://github.com/rancher/storage/commit/08a47cbd0d721e4caa5c011b4b6132920cefdc3a is exactly what I'm talking about. mountdest() and mount_default_share() are almost the same function, but there's 2 copies of it and it is error prone. My PR actually already fixed that.
